src/statistics.sh

```diff
@@ -195,14 +195,14 @@ function sec_to_formatted_date()
 # Note: This function relies on a global variable named shared_data.
 function print_basic_data()
 {
-  local line=" Total Max Min Average\n"
-  for option in "${!statistics_opt[@]}"; do
-    [[ -z "${shared_data[$option]}" ]] && continue
+  local header_format="%20s %4s %8s %12s\n"
+  local row_format="%-14s %5d %s %s %s\n"
 
-    line="${line}${option^} ${shared_data[$option]}\n"
+  printf "$header_format" Total Max Min Average
+  for option in "${statistics_opt[@]}"; do
+    [[ -z "${shared_data[$option]}" ]] && continue
+    printf "$row_format" ${option^} ${shared_data[$option]}
```

Comment:
Any particular reason for not add double-quotes around `${option^}` and `${shared_data[$option]}`?

Reply:
I could add it to `${option^}`, it slipped my mind. Thanks for pointing that out.
As for `${shared_data[$option]}`, if I do double-quote it, then it won't work. `printf` is expecting a string, a number, and then three other strings (that is the row format I specified above). From my understanding, if I double-quote `${shared_data[$option]}`, then it will be passed as one whole thing. If I don't, then it will pass an element of the array at a time. Does it make sense?

Reply:
Yes, make sense. Thanks